adobe-esign-connector/processes/connector/TransientDocuments.p.json

```diff
@@ -167,5 +167,247 @@
+                "out.error" : "#error"
+              }
+            }
+          },
+          "visual" : {
+            "at" : { "x" : 416, "y" : 402 }
+          },
+          "connect" : { "id" : "f20", "to" : "f15", "via" : [ { "x" : 776, "y" : 400 } ] }
+        } ],
+      "connect" : { "id" : "f27", "to" : "f25" }
+    }, {
+      "id" : "f21",
+      "type" : "Script",
+      "name" : "pop file from list",
+      "config" : {
+        "output" : {
+          "code" : "in.file = in.files.removeGet(0);"
+        }
+      },
+      "visual" : {
+        "at" : { "x" : 224, "y" : 360 }
+      },
+      "connect" : { "id" : "f18", "to" : "f16" }
+    }, {
+      "id" : "f23",
+      "type" : "Alternative",
+      "name" : "files size > 0",
+      "visual" : {
+        "at" : { "x" : 672, "y" : 360 },
+        "labelOffset" : { "x" : 14, "y" : 34 }
+      },
+      "connect" : [
+        { "id" : "f26", "to" : "f21", "via" : [ { "x" : 672, "y" : 304 }, { "x" : 224, "y" : 304 } ], "label" : {
+            "name" : "yes",
+            "segment" : 0.5,
+            "offset" : { "x" : 1, "y" : -6 }
+          }, "condition" : "in.files.size() > 0" },
+        { "id" : "f19", "to" : "f15", "label" : {
+            "name" : "no"
+          } }
+      ]
+    }, {
+      "id" : "f25",
+      "type" : "Script",
+      "name" : "add id result",
+      "config" : {
+        "output" : {
+          "code" : "in.ids.add(in.id);"
+        }
+      },
+      "visual" : {
+        "at" : { "x" : 544, "y" : 360 }
+      },
+      "connect" : { "id" : "f24", "to" : "f23" }
+    }, {
+      "id" : "f28",
+      "type" : "CallSubStart",
+      "name" : "uploadDocuments(List<UploadWrapper)",
+      "config" : {
+        "callSignature" : "uploadDocuments",
+        "input" : {
+          "params" : [
+            { "name" : "uploadWrappers", "type" : "java.util.List<com.axonivy.connector.adobe.esign.connector.rest.UploadWrapper>", "desc" : "file to upload" }
+          ],
+          "map" : {
+            "out.uploadWrappers" : "param.uploadWrappers"
+          }
+        },
+        "result" : {
+          "params" : [
+            { "name" : "ids", "type" : "java.util.List<String>", "desc" : "ids of uploaded files" },
+            { "name" : "error", "type" : "ch.ivyteam.ivy.bpm.error.BpmError", "desc" : "holds error if one occurs, empty otherwise" }
+          ],
+          "map" : {
+            "result.ids" : "in.#ids",
+            "result.error" : "in.#error"
+          }
+        }
+      },
+      "visual" : {
+        "at" : { "x" : 96, "y" : 512 }
+      },
+      "connect" : { "id" : "f39", "to" : "f32" }
+    }, {
+      "id" : "f29",
+      "type" : "CallSubEnd",
+      "visual" : {
+        "at" : { "x" : 776, "y" : 512 }
+      }
+    }, {
+      "id" : "f32",
+      "type" : "Script",
+      "name" : "pop file from list",
+      "config" : {
+        "output" : {
+          "code" : "in.upload = in.uploadWrappers.removeGet(0);"
```

Comment:
please check null or empty before modify the collection to avoid outofbound exception

Reply:
should be addressed already